### PR TITLE
UXUI-245 - Initial contributing commit based on atom contributing.md

### PR DESCRIPTION
**Pull request by jodiedoubleday**
#### What does this PR do?
- As UI-Toolkit is open source, if we are going to have people contribute to it then we should have clear guidelines on how people can contribute to the repo.

#### How should this be tested?
- Have a read, look for any spelling errors etc. If you see anything you like or dislike, let me know.
- Any extra stuff you think should be added then stick on your own PR after this one has gone live.

#### What gif best describes how you feel about this work?
Contributing

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
+# Contributing to UI-Toolkit
+
+A BIG thank you from everyone using the UI-Toolkit for taking the time to contribute! :clap:
+
+The following is a set of guidelines for contributing to the UI-Tookit and its components,
+which are hosted in the [Holiday Extras Organization](https://github.com/holidayextras) on GitHub.
+These are just guidelines, not rules, use your best judgment and feel free to
+propose changes to this document in a pull request.
+
+This guide will be constently updated.
```
Comment (cameronviner) [inline CONTRIBUTING.md]: Typo here, should be: constantly

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+## New Components
+* You can creat new components easily using the
```
Comment (cameronviner) [inline CONTRIBUTING.md]: Typo here as well: create

**Comment by cameronviner** [comment]
Couple of typos but this looks good :+1:

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
+# Contributing to UI-Toolkit
+
+A BIG thank you from everyone using the UI-Toolkit for taking the time to contribute! :clap:
+
+The following is a set of guidelines for contributing to the UI-Toolkit and its components,
+which are hosted in the [Holiday Extras Organization](https://github.com/holidayextras) on GitHub.
```
Comment (lukefrake) [inline CONTRIBUTING.md]: Doesn't feel necessary to mention HX here as github has nice URL structure

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+* Please choose a name that describes the component i.e `button`
+* When adding a class to the template, please prefix with `component-` so when in use it doesn't conflict with existing
+  components
+
+## Submitting Issues
+
+* You can create a [new issue](https://github.com/holidayextras/ui-toolkit/issues/new), but
+  before doing that please read the notes below on submitting issues,
+  and include as many details as possible with your report.
+* Include screenshots and animated GIFs whenever possible; they are immensely
```
Comment (lukefrake) [inline CONTRIBUTING.md]: Mention [Quickcast](quickcast.io)? - Super useful!

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+* Include screenshots and animated GIFs in your pull request whenever possible.
+* Include thoughtfully-worded, well-structured
+  [Jest](https://facebook.github.io/jest/) specs in the `__tests__` folder in area for developement, if the directory
+  doesn't exist, please add one. Run them using `jest`. See the [Specs Styleguide](#specs-styleguide) below.
+
+### Linting
+We use [ESLint](http://eslint.org/) to ensure consistency in our repo.
+Please ensure you:
+
+* Illiminate excess white space.
```
Comment (lukefrake) [inline CONTRIBUTING.md]: Is this not built into ESLint? If so does it need to be said again?

Comment (lukefrake) [inline CONTRIBUTING.md]: Also, do you use something like travis at build? If you do I think mention that here

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+* End files with a newline.
+
+## Git Commit Messages
+
+* Limit the first line to 72 characters or less
+* Reference issues and pull requests liberally
+* Consider starting the commit message with an applicable emoji:
```
Comment (lukefrake) [inline CONTRIBUTING.md]: :+1: love this

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+    * :racehorse: `:racehorse:` when improving performance
+    * :memo: `:memo:` when writing docs
+    * :fire: `:fire:` when removing code or files
+    * :lock: `:lock:` when dealing with security
+    * :arrow_up: `:arrow_up:` when upgrading dependencies
+    * :arrow_down: `:arrow_down:` when downgrading dependencies
+
+## Specs Styleguide
+
+- Include thoughtfully-worded, well-structured
```
Comment (lukefrake) [inline CONTRIBUTING.md]: This line is a duplicate of line 35, any way to merge into one and don't repeat??

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+describe 'a dog', ->
+ it 'barks', ->
+ # spec here
+ describe 'when the dog is happy', ->
+  it 'wags its tail', ->
+  # spec here
+```
+
+--
+Based on the wonderful Contribution.md file from the guys and girls at [atom](https://github.com/atom/atom)
```
Comment (lukefrake) [inline CONTRIBUTING.md]: :+1:

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+## Specs Styleguide
+
+- Include thoughtfully-worded, well-structured
+  [Jest](https://facebook.github.io/jest/) specs in the `__test__` folder.
+- treat `describe` as a noun or situation.
+- treat `it` as a statement about state or how an operation changes state.
+
+### Example
+
+```coffee
```
Comment (lukefrake) [inline CONTRIBUTING.md]: Are we phasing out coffee script?

Comment (hpoom) [inline CONTRIBUTING.md]: Yes we are phasing out coffee

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+
+* You can create a [new issue](https://github.com/holidayextras/ui-toolkit/issues/new), but
+  before doing that please read the notes below on submitting issues,
+  and include as many details as possible with your report.
+* Include screenshots and animated GIFs whenever possible; they are immensely
+  helpful.
+* Include the behavior you expected and other places you've seen that behavior
+* Perform a [cursory search](https://github.com/holidayextras/ui-toolkit/issues)
+  to see if a similar issue has already been submitted.
+* Please setup a [profile picture](https://help.github.com/articles/how-do-i-set-up-my-profile-picture)
```
Comment (hpoom) [inline CONTRIBUTING.md]: Some of our own HX Developers have not done this yet. We should get our own house in order first. But yes love this point.

File: CONTRIBUTING.md (hunk 8 above)
Comment (hpoom) [inline CONTRIBUTING.md]: :+1:

File: CONTRIBUTING.md (hunk 9 above)
Comment (Joezo) [inline CONTRIBUTING.md]: not too mention this repo doesn't use coffee, probably worth swapping this out for an example in this repo jodie :)

File: CONTRIBUTING.md (hunk 4 above)
Comment (jodiedoubleday) [inline CONTRIBUTING.md]: Link is dead

File: CONTRIBUTING.md (hunk 5 above)
Comment (jodiedoubleday) [inline CONTRIBUTING.md]: I think its good to mention the simple things people can do, before running it through a linter.

File: CONTRIBUTING.md (hunk 9 above)
Comment (jodiedoubleday) [inline CONTRIBUTING.md]: I want to stay away form a specific example from the repo, and use more of a guide.

File: CONTRIBUTING.md (hunk 5 above)
Comment (lukefrake) [inline CONTRIBUTING.md]: I see where your coming from, for me would be better linking to the [.eslintrc](https://github.com/holidayextras/ui-toolkit/blob/master/.eslintrc) as that will be constantly up to date.

If I was to read this list, I would think thats an exhaustive list of everything you lint against.

Up to you though, just my opinion :)

Comment (hxpaul) [inline CONTRIBUTING.md]: Eliminate

File: CONTRIBUTING.md [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+
+The following is a set of guidelines for contributing to the UI-Toolkit and its components,
+These are just guidelines, not rules, use your best judgment and feel free to
+propose changes to this document in a pull request.
+
+This guide will be constantly updated.
+
+## New Components
+* You can create new components easily using the
+[Component Generator](https://github.com/holidayextras/generator-ui-component)
```
Comment (hxpaul) [inline CONTRIBUTING.md]: might as well make these links relative

Comment (hpoom) [inline CONTRIBUTING.md]: Yes all links in GitHub markdown files should be relative

https://github.com/blog/1395-relative-links-in-markup-files

Comment (jodiedoubleday) [inline CONTRIBUTING.md]: This leads to a different repository so can't be relative in case it is forked.

Comment (lukefrake) [inline CONTRIBUTING.md]: How do you mean?
[Component Generator](/holidayextras/generator-ui-component) should always resolve forked or not?

Comment (hpoom) [inline CONTRIBUTING.md]: OK fair point on that one. The others though back to the same repo can be relative.

**Pull request merged**